python/ray/data/tests/test_streaming_executor.py

```diff
@@ -720,6 +727,54 @@ def patched_get_outupt_blocking(*args, **kwargs):
     assert isinstance(error, KeyboardInterrupt), error
 
 
+def test_execution_callbacks_topology_arg(tmp_path, restore_data_context):
+    """Test the topology arg in ExecutionCallback."""
+
+    topo = None
+
+    class CustomExecutionCallback(ExecutionCallback):
+        def after_execution_succeeds(self, topology: Topology):
+            nonlocal topo
+            topo = topology
+
+    input_path = tmp_path / "input"
+    os.makedirs(input_path)
+    output_path = tmp_path / "output"
+
+    ctx = DataContext.get_current()
+    callback = CustomExecutionCallback()
+    add_execution_callback(callback, ctx)
+    ds = ray.data.read_parquet(input_path)
+
+    def udf(row):
+        return row
+
+    ds = ds.map(udf)
+
+    ds = ds.write_parquet(output_path)
+
+    assert topo is not None
+    assert len(topo) == 2
+    physical_ops = list(topo.keys())
+    assert isinstance(physical_ops[0], InputDataBuffer)
+    assert isinstance(physical_ops[1], MapOperator)
+    logical_ops = physical_ops[1]._logical_operators
+
+    assert len(logical_ops) == 3
+    assert isinstance(logical_ops[0], Read)
+    datasource = logical_ops[0]._datasource
+    assert isinstance(datasource, ParquetDatasource)
+    assert datasource.unresolved_path == input_path
+
+    assert isinstance(logical_ops[1], MapRows)
+    assert logical_ops[1]._fn == udf
+
+    assert isinstance(logical_ops[2], Write)
+    datasink = logical_ops[2]._datasink_or_legacy_datasource
+    assert isinstance(datasink, ParquetDatasink)
+    assert datasink.unresolved_path == output_path
```

Comment:
I think if other people read this test they'd fine it weird that we're asserting against `unresolved_path`, since it isn't obviously related to the `topology` arg. Not sure if it'd make sense to add a comment given the context, though

Reply:
added a comment